tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -148,13 +149,50 @@ append_ec_for_seqnum(PlannerInfo *root, CompressionInfo *info, SortInfo *sort_in
 	return newec;
 }
 
+static EquivalenceClass *
+append_ec_for_metadata_col(PlannerInfo *root, CompressionInfo *info, Var *var, PathKey *pk)
+{
+	MemoryContext oldcontext = MemoryContextSwitchTo(root->planner_cxt);
+	EquivalenceMember *em = makeNode(EquivalenceMember);
+
+	em->em_expr = (Expr *) var;
+	em->em_relids = bms_make_singleton(info->compressed_rte->relid);
+	em->em_is_const = false;
+	em->em_is_child = false;
+	em->em_datatype = var->vartype;
+	EquivalenceClass *ec = makeNode(EquivalenceClass);
+	ec->ec_opfamilies = pk->pk_eclass->ec_opfamilies;
+	ec->ec_collation = pk->pk_eclass->ec_collation;
+	ec->ec_members = list_make1(em);
+	ec->ec_sources = list_copy(pk->pk_eclass->ec_sources);
+	ec->ec_derives = list_copy(pk->pk_eclass->ec_derives);
+	ec->ec_relids = bms_make_singleton(info->compressed_rel->relid);
+	ec->ec_has_const = pk->pk_eclass->ec_has_const;
+	ec->ec_has_volatile = pk->pk_eclass->ec_has_volatile;
+#if PG16_LT
+	ec->ec_below_outer_join = pk->pk_eclass->ec_below_outer_join;
+#endif
+	ec->ec_broken = pk->pk_eclass->ec_broken;
+	ec->ec_sortref = pk->pk_eclass->ec_sortref;
+	ec->ec_min_security = pk->pk_eclass->ec_min_security;
+	ec->ec_max_security = pk->pk_eclass->ec_max_security;
+	ec->ec_merged = pk->pk_eclass->ec_merged;
+	MemoryContextSwitchTo(oldcontext);
+	root->eq_classes = lappend(root->eq_classes, ec);
```

Comment:
Should this append also be on the planner root memory context?

Reply:
Yeah, I think you are right. Good catch, fixing.